platform/linux-generic/pktio/null.c

```diff
@@ -0,0 +1,207 @@
+	return 0;
+}
+
+static int null_recv(pktio_entry_t *pktio_entry ODP_UNUSED,
+		     int index ODP_UNUSED, odp_packet_t pkt_table[] ODP_UNUSED,
+		     int len ODP_UNUSED)
+{
+	return 0;
+}
+
+static int null_fd_set(pktio_entry_t *pktio_entry ODP_UNUSED,
+		       int index ODP_UNUSED, fd_set *readfds ODP_UNUSED)
+{
+	return 0;
+}
+
+static int null_recv_tmo(pktio_entry_t *pktio_entry ODP_UNUSED,
+			 int index ODP_UNUSED,
+			 odp_packet_t pkt_table[] ODP_UNUSED,
+			 int num ODP_UNUSED, uint64_t usecs)
+{
+	struct timeval timeout;
+	int maxfd = -1;
+	fd_set readfds;
+
+	timeout.tv_sec = usecs / 1000 / 1000;
+	timeout.tv_usec = usecs % (1000 * 1000);
+	FD_ZERO(&readfds);
+
+	select(maxfd + 1, &readfds, NULL, NULL,
+	       usecs == ODP_PKTIN_WAIT ? NULL : &timeout);
+
+	return 0;
+}
+
+static int null_recv_mq_tmo(pktio_entry_t *pktio_entry[] ODP_UNUSED,
+			    int index[] ODP_UNUSED, int num_q ODP_UNUSED,
+			    odp_packet_t pkt_table[] ODP_UNUSED,
+			    int num ODP_UNUSED, unsigned *from ODP_UNUSED,
+			    uint64_t usecs)
+{
+	struct timeval timeout;
+	int maxfd = -1;
+	fd_set readfds;
+
+	timeout.tv_sec = usecs / 1000 / 1000;
+	timeout.tv_usec = usecs % (1000 * 1000);
+
+	FD_ZERO(&readfds);
+
+	select(maxfd + 1, &readfds, NULL, NULL,
+	       usecs == ODP_PKTIN_WAIT ? NULL : &timeout);
+
+	return 0;
+}
+
+static int null_send(pktio_entry_t *pktio_entry ODP_UNUSED,
+		     int index ODP_UNUSED, const odp_packet_t pkt_table[],
+		     int num)
+{
+	odp_packet_free_multi(pkt_table, num);
+
+	return num;
+}
+
+#define PKTIO_NULL_MTU (64 * 1024)
+
+static uint32_t null_mtu_get(pktio_entry_t *pktio_entry ODP_UNUSED)
+{
+	return PKTIO_NULL_MTU;
+}
+
+static const char null_mac[] = {0x02, 0xe9, 0x34, 0x80, 0x73, 0x05};
+
+static int null_mac_addr_get(pktio_entry_t *pktio_entry ODP_UNUSED,
+			     void *mac_addr)
+{
+	memcpy(mac_addr, null_mac, ETH_ALEN);
+	return ETH_ALEN;
+}
+
+static int null_promisc_mode_set(pktio_entry_t *pktio_entry, odp_bool_t enable)
+{
+	pktio_entry->s.pkt_null.promisc = !!enable;
+	return 0;
+}
+
+static int null_promisc_mode_get(pktio_entry_t *pktio_entry)
+{
+	return pktio_entry->s.pkt_null.promisc;
+}
+
+static int null_capability(pktio_entry_t *pktio_entry ODP_UNUSED,
+			   odp_pktio_capability_t *capa)
+{
+	memset(capa, 0, sizeof(odp_pktio_capability_t));
+
+	capa->max_input_queues  = PKTIO_MAX_QUEUES;
+	capa->max_output_queues = PKTIO_MAX_QUEUES;
+	capa->set_op.op.promisc_mode = 1;
+
+	odp_pktio_config_init(&capa->config);
+	capa->config.pktin.bit.ts_all = 1;
+	capa->config.pktin.bit.ts_ptp = 1;
+	return 0;
+}
+
+static int null_inqueues_config(pktio_entry_t *pktio_entry ODP_UNUSED,
+				const odp_pktin_queue_param_t *p ODP_UNUSED)
+{
+	return 0;
+}
+
+static int null_outqueues_config(pktio_entry_t *pktio_entry ODP_UNUSED,
+				 const odp_pktout_queue_param_t *p ODP_UNUSED)
+{
+	return 0;
+}
+
+const pktio_if_ops_t null_pktio_ops = {
+	.name = "null",
+	.print = NULL,
+	.init_global = NULL,
```

Comment:
I think other pktio have debug print that pktio is available on early init. Please keep the same style.

Reply:
Done, the same style is used now in null pktio as well.